staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -147,24 +147,24 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, namespace, na
 		if patch == nil {
 			patch, err = p.buildStrategicMergeFromBuiltins(p.Mapping.GroupVersionKind, original, modified, current)
 			if err != nil {
-				return nil, nil, errors.Wrapf(err, createPatchErrFormat, original, modified, current)
+				return false, nil, errors.Wrapf(err, createPatchErrFormat, original, modified, current)
 			}
 		}
 	}
 
 	if string(patch) == "{}" {
-		return patch, obj, nil
+		return false, obj, nil
 	}
 
 	if p.ResourceVersion != nil {
 		patch, err = addResourceVersion(patch, *p.ResourceVersion)
 		if err != nil {
-			return nil, nil, errors.Wrap(err, "Failed to insert resourceVersion in patch")
+			return true, nil, errors.Wrap(err, "Failed to insert resourceVersion in patch")
 		}
 	}
 
 	patchedObj, err := p.Helper.Patch(namespace, name, patchType, patch, nil)
-	return patch, patchedObj, err
+	return true, patchedObj, errors.Wrapf(err, "applying patch:\n%v\nto:\n%v\nfor:", patch, obj)
```

Comment:
In reality, this is the only place we can embed patch diff into the error message, instead doing this in here https://github.com/kubernetes/kubernetes/pull/110829/files#diff-0d73c392875ae1876a4e74dccf25726c83ad93a850aa2fbd9560438efc467f9aL629